crates/extensions/c8y_mapper_ext/src/operations/handler.rs

```diff
@@ -206,51 +260,27 @@ impl OperationHandler {
         topics
     }
 }
-
 struct RunningOperation {
     handle: tokio::task::JoinHandle<()>,
-    tx: tokio::sync::mpsc::UnboundedSender<OperationMessage>,
+    status: String,
 }
 
-impl RunningOperation {
-    /// Spawns a task that handles the operation.
-    ///
-    /// The task handles a single operation with a given command id, and via a channel it receives
-    /// operation state changes (if any) to drive an operation to completion.
-    fn spawn(operation: Arc<OperationContext>) -> Self {
-        let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
-
-        let handle = tokio::spawn(async move {
-            while let Some(message) = rx.recv().await {
-                if let UpdateStatus::Terminated = operation.update(message).await {
-                    break;
-                }
-            }
-        });
+// TODO: logic of which status transitions are valid should be defined in tedge_api and be
+// considered together with custom statuses of custom workflows
+fn is_operation_status_transition_valid(previous: &str, next: &str) -> bool {
+    #[allow(clippy::match_like_matches_macro)]
+    match (previous, next) {
+        // not really a transition but true for simplicity
+        (prev, next) if prev == next => true,
```

Comment:
```suggestion
        (prev, next) if prev == next => false,
```
Allowing duplicate state transitions is problematic when we use SmartREST. For example, if the mapper processes two `executing` responses back-to-back, the first mapped `501` message will properly change the status of the original request. But if we send another `501` message for the duplicated `executing` response, then it might wrongly change the status of the next operation of the same type, if multiple requests of the same type are queued together on the cloud. This is a side effect of the id-less design of SmartREST that does not target a specific operation id, but always tries to change that status of the operations in the order in which they were triggered. So, we should just skip mapping duplicate state transitions. Fortunately, the check in line no. 174 is already doing that, which bypasses this case here. But, it's still better to keep it consistent here as well.

Reply:
Yeah, it depends what we mean by "transition". Here I meant more like an identity transition of the status on the operation itself, not really related to reporting of the transition to Cumulocity. As the TODO says, if it were declared in tedge_api it would be a bit more clear, but here it looks like the logic is duplicated, which I agree with.
Still, for consistency's sake, changed the case to false in 46f1f2a090.